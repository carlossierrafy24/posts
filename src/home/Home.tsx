import React from 'react';
import { Link } from 'react-router-dom';
import { useAppDispatch, useAppSelector } from '../state/hooks/hooks';
import { logout } from '../state/slices/UserSlice';

const Home: React.FC = (): JSX.Element => {
    const user = useAppSelector((state) => state.user);
    const dispatch = useAppDispatch();

    const handleLogout = () => {
        user.isAuthenticated && dispatch(logout());
    };

    return (
        <div>
            <nav className="flex justify-between items-center py-0 px-5 fixed h-16 bg-indigo-500 top-0 left-0 right-0 z-1">
                <div className="flex justify-center items-center">
                    <div className="text-2xl font-bold text-white mr-3">Home</div>
                    <div>
                        {user.isAuthenticated && (
                            <Link to="/dashboard" className="text-white text-xl font-semibold hover:bg-indigo-300 hover:rounded-md py-1 px-2 mr-2">
                                Dashboard
                            </Link>
                        )}
                    </div>
                </div>
                <div>
                    <div className="text-center">
                        <h5 className="text-lg text-white mr-2">{user.isAuthenticated && user.name}</h5>
                        <Link onClick={handleLogout} to={user.isAuthenticated ? '/' : '/signin'} className="text-white text-xl font-semibold hover:bg-indigo-300 hover:rounded-md px-2 py-1">
                            {user.isAuthenticated ? 'Logout ⤵️' : 'Signin ⤴️'}
                        </Link>
                    </div>
                </div>
            </nav>
            <div className="bg-white overflow-hidden mt-8 ml-6 flex justify-center items-center">
                <div className=" pb-72 sm:pt-24 sm:pb-40 lg:pt-40 lg:pb-48">
                    <div className="relative max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 sm:static">
                        <div className="sm:max-w-lg">
                            <h1 className="text-4xl font font-extrabold tracking-tight text-gray-900 sm:text-6xl">Posts & Comments</h1>
                            <p className="mt-4 text-xl text-gray-500">
                                React-Redux-Typescript application developed by{' '}
                                <a href="https://github.com/carlossierrafy24" className="text-blue-500">
                                    carlossierrafy24
                                </a>
                            </p>
                        </div>
                        <div>
                            <div className="mt-8">
                                {!user.isAuthenticated && (
                                    <Link
                                        to="/signin"
                                        className="inline-block text-center border border-transparent rounded-md py-3 px-8 font-medium text-white bg-cyan-800 hover:bg-cyan-600 focus:outline-none">
                                        Sign In ➡️
                                    </Link>
                                )}
                            </div>
                        </div>
                    </div>
                </div>
                <div>
                    <div aria-hidden="true" className="pointer-events-none">
                        <img className="mx-auto" src="./image.png" width={'70%'} alt="stack" />
                    </div>
                </div>
            </div>
        </div>
    );
};

export default Home;
