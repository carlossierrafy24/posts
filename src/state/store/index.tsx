import { configureStore } from '@reduxjs/toolkit';
import { persistStore, persistReducer, FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import userReducer from '../slices/UserSlice';

const persistConfig = {
    key: 'root',
    version: 1,
    storage
};

const rootReducer = persistReducer(persistConfig, userReducer);

const store = configureStore({
    reducer: { user: rootReducer },
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
            serializableCheck: {
                ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER]
            }
        })
});
export let persistor = persistStore(store);
export default store;

export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
