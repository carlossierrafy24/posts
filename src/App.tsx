import React from 'react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Home from './home/Home';
import SignIn from './signIn/SignIn';
import './index.css';
import Dashboard from './dashboard/Dashboard';
import ProtectedRoute from './components/ProtectedRoute';
import Error404 from './components/404/404';

const App: React.FC = function () {
    document.title = 'Home';
    return (
        <div>
            <Router>
                <Routes>
                    <Route path="/" element={<Home />} />
                    <Route path="/signin" element={<SignIn />} />
                    <Route
                        path="/dashboard"
                        element={
                            <ProtectedRoute>
                                <Dashboard />
                            </ProtectedRoute>
                        }
                    />
                    <Route path="*" element={<Error404 />} />
                </Routes>
            </Router>
        </div>
    );
};
export default App;
