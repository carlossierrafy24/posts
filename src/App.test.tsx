import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../src/state/store';
import App from './App';
import Home from './home/Home';
import { BrowserRouter as Router } from 'react-router-dom';

test('renders learn react link', () => {
    render(
        <Provider store={store}>
            <App />
        </Provider>
    );

    expect(screen.getByText(/Home/i)).toBeInTheDocument();
    expect(document.title).toBe('Home');
    expect(
        render(
            <Provider store={store}>
                <Router>
                    <Home />
                </Router>
            </Provider>
        )
    ).toMatchSnapshot();
});
