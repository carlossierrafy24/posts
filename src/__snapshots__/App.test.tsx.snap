// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <nav
            class="flex justify-between items-center py-0 px-5 fixed h-16 bg-indigo-500 top-0 left-0 right-0 z-1"
          >
            <div
              class="flex justify-center items-center"
            >
              <div
                class="text-2xl font-bold text-white mr-3"
              >
                Home
              </div>
              <div>
                <a
                  class="text-white text-xl font-semibold hover:bg-indigo-300 hover:rounded-md py-1 px-2 mr-2"
                  href="/dashboard"
                >
                  Dashboard
                </a>
              </div>
            </div>
            <div>
              <div
                class="text-center"
              >
                <h5
                  class="text-lg text-white mr-2"
                />
                <a
                  class="text-white text-xl font-semibold hover:bg-indigo-300 hover:rounded-md px-2 py-1"
                  href="/"
                >
                  Logout ⤵️
                </a>
              </div>
            </div>
          </nav>
          <div
            class="bg-white overflow-hidden mt-8 ml-6 flex justify-center items-center"
          >
            <div
              class=" pb-72 sm:pt-24 sm:pb-40 lg:pt-40 lg:pb-48"
            >
              <div
                class="relative max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 sm:static"
              >
                <div
                  class="sm:max-w-lg"
                >
                  <h1
                    class="text-4xl font font-extrabold tracking-tight text-gray-900 sm:text-6xl"
                  >
                    Posts & Comments
                  </h1>
                  <p
                    class="mt-4 text-xl text-gray-500"
                  >
                    React-Redux-Typescript application developed by
                     
                    <a
                      class="text-blue-500"
                      href="https://github.com/carlossierrafy24"
                    >
                      carlossierrafy24
                    </a>
                  </p>
                </div>
                <div>
                  <div
                    class="mt-8"
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                aria-hidden="true"
                class="pointer-events-none"
              >
                <img
                  alt="stack"
                  class="mx-auto"
                  src="./image.png"
                  width="70%"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        <nav
          class="flex justify-between items-center py-0 px-5 fixed h-16 bg-indigo-500 top-0 left-0 right-0 z-1"
        >
          <div
            class="flex justify-center items-center"
          >
            <div
              class="text-2xl font-bold text-white mr-3"
            >
              Home
            </div>
            <div>
              <a
                class="text-white text-xl font-semibold hover:bg-indigo-300 hover:rounded-md py-1 px-2 mr-2"
                href="/dashboard"
              >
                Dashboard
              </a>
            </div>
          </div>
          <div>
            <div
              class="text-center"
            >
              <h5
                class="text-lg text-white mr-2"
              />
              <a
                class="text-white text-xl font-semibold hover:bg-indigo-300 hover:rounded-md px-2 py-1"
                href="/"
              >
                Logout ⤵️
              </a>
            </div>
          </div>
        </nav>
        <div
          class="bg-white overflow-hidden mt-8 ml-6 flex justify-center items-center"
        >
          <div
            class=" pb-72 sm:pt-24 sm:pb-40 lg:pt-40 lg:pb-48"
          >
            <div
              class="relative max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 sm:static"
            >
              <div
                class="sm:max-w-lg"
              >
                <h1
                  class="text-4xl font font-extrabold tracking-tight text-gray-900 sm:text-6xl"
                >
                  Posts & Comments
                </h1>
                <p
                  class="mt-4 text-xl text-gray-500"
                >
                  React-Redux-Typescript application developed by
                   
                  <a
                    class="text-blue-500"
                    href="https://github.com/carlossierrafy24"
                  >
                    carlossierrafy24
                  </a>
                </p>
              </div>
              <div>
                <div
                  class="mt-8"
                />
              </div>
            </div>
          </div>
          <div>
            <div
              aria-hidden="true"
              class="pointer-events-none"
            >
              <img
                alt="stack"
                class="mx-auto"
                src="./image.png"
                width="70%"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="flex justify-between items-center py-0 px-5 fixed h-16 bg-indigo-500 top-0 left-0 right-0 z-1"
      >
        <div
          class="flex justify-center items-center"
        >
          <div
            class="text-2xl font-bold text-white mr-3"
          >
            Home
          </div>
          <div>
            <a
              class="text-white text-xl font-semibold hover:bg-indigo-300 hover:rounded-md py-1 px-2 mr-2"
              href="/dashboard"
            >
              Dashboard
            </a>
          </div>
        </div>
        <div>
          <div
            class="text-center"
          >
            <h5
              class="text-lg text-white mr-2"
            />
            <a
              class="text-white text-xl font-semibold hover:bg-indigo-300 hover:rounded-md px-2 py-1"
              href="/"
            >
              Logout ⤵️
            </a>
          </div>
        </div>
      </nav>
      <div
        class="bg-white overflow-hidden mt-8 ml-6 flex justify-center items-center"
      >
        <div
          class=" pb-72 sm:pt-24 sm:pb-40 lg:pt-40 lg:pb-48"
        >
          <div
            class="relative max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 sm:static"
          >
            <div
              class="sm:max-w-lg"
            >
              <h1
                class="text-4xl font font-extrabold tracking-tight text-gray-900 sm:text-6xl"
              >
                Posts & Comments
              </h1>
              <p
                class="mt-4 text-xl text-gray-500"
              >
                React-Redux-Typescript application developed by
                 
                <a
                  class="text-blue-500"
                  href="https://github.com/carlossierrafy24"
                >
                  carlossierrafy24
                </a>
              </p>
            </div>
            <div>
              <div
                class="mt-8"
              />
            </div>
          </div>
        </div>
        <div>
          <div
            aria-hidden="true"
            class="pointer-events-none"
          >
            <img
              alt="stack"
              class="mx-auto"
              src="./image.png"
              width="70%"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
